### PR TITLE
Defined cmath with the IRDL from xDSL + added tests

### PR DESCRIPTION
**Pull request by purvi-H**
Wrote and registered the cmath dialect + tested its operations.

### CODE REVIEW

File: tests/filecheck/.lit_test_times.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+7.216287e-02 arith_invalid.xdsl
+2.446129e-01 arith_ops.xdsl
+1.118841e-01 builtin_types.xdsl
+2.239571e-01 cf_ops.xdsl
+2.121851e-01 escaped_characters.xdsl
+1.508312e-01 func_ops.xdsl
+2.111490e-01 memref_ops.xdsl
+5.006790e-06 mlir-conversion/ops.xdsl
+2.215171e-01 scf_ops.xdsl
+9.212685e-02 cmath.xdsl
```
Comment (tobiasgrosser) [inline tests/filecheck/.lit_test_times.txt:11]: Does this need to be added?

**Review by tobiasgrosser — COMMENTED** [review]
Very cool. Thank you for this PR.

File: src/xdsl/dialects/cmath.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+
+from __future__ import annotations
+from dataclasses import dataclass
+
+from xdsl.dialects.builtin import * 
+
+from xdsl.irdl import *
+from xdsl.ir import *
+from xdsl.dialects.irdl import *
```
Comment (math-fehr) [inline src/xdsl/dialects/cmath.py:9]: Can you remove the unused imports (irdl, func, and arith)?

File: src/xdsl/dialects/cmath.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+from xdsl.dialects.arith import *
+from xdsl.dialects.builtin import *
+
+# Used for cyclic dependencies in type hints
+if TYPE_CHECKING:
+    from xdsl.parser import Parser
+    from xdsl.printer import Printer
+
+@dataclass
+class Cmath:
```
Comment (math-fehr) [inline src/xdsl/dialects/cmath.py:20]: Could you rename it to `CMath`?

File: src/xdsl/dialects/cmath.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    # TODO replace with trait
+    def verify_(self) -> None:
+        if self.operands.typ != self.results.typ:
+            raise VerifyException(
+                "expect all input and output types to be equal")
+
+@irdl_op_definition
+class Mul(Operation):
+    name: str = "irdl.mul"
+    data: ParameterDef[Float64Type | Float32Type]
```
Comment (math-fehr) [inline src/xdsl/dialects/cmath.py:50]: Can you remove this line?

File: tests/filecheck/cmath.xdsl [hunk 5]
```diff
@@ -0,0 +1,36 @@
+// RUN: xdsl-opt %s | xdsl-opt | filecheck %s
```
Comment (math-fehr) [inline tests/filecheck/cmath.xdsl:1]: So this file is the cmath dialect, but written as a program with the IRDL dialect.
Could you remove this file for a subsequent PR? And instead have a file that test CMath.
So for instance something like https://github.com/opencompl/dyn-dialect/blob/main/test/Dyn/test-cmath.mlir, but using an xDSL syntax

**Review by math-fehr — CHANGES_REQUESTED** [review]
Could you fix the comments we sent?
Also, could you format your files using yapf, so it passes the CI?

**Pull request merged**